File: generator.rb
generator = Fiber.new do |a|
  puts a
  b = Fiber.yield 1
  puts b
  c = Fiber.yield 2
  puts c
end

puts generator.resume 'a'
puts generator.resume 'b'
puts generator.resume 'c'
puts generator.resume 'd'

File: generator.js
function* generatorFunc(a) {
  console.log(a);
  var b = yield 1;
  console.log(b);
  var c = yield 2;
  console.log(c);
}

var generator = generatorFunc('a');
console.log(generator.next('b'));
console.log(generator.next('c'));
console.log(generator.next('d'));
